### ISSUE TITLE
Change Request: build & publish cjs bundle

### ISSUE THREAD

**Issue by aladdin-add** [eslint/markdown#307]
### Environment

ESLint version: v9.16.0
@eslint/markdown version: v6.2.1
Node version:  v22.11.0
npm version: n/a
Operating System: n/a

### What problem do you want to solve?

This package only published esm, no cjs. 
![Image](https://github.com/user-attachments/assets/7b8bc7c7-e11d-48be-8a6a-352516d9d658)

For projects with `pkg.type = "commonjs"` (the node.js default) (e.g. `eslint/eslint`), it's not possible to use it in `eslint.config.js`.

```js
const markdown = require("@eslint/markdown"); // ❌Error! 
```

the workaround is to convert the config file to `eslint.config.mjs`.

### What do you think is the correct solution?

add rollup config to build cjs, and publish it to npm.

### Participation

- [x] I am willing to submit a pull request for this change.

### Additional comments

Technically, it's a feature request; not a bug - as the docs only mentioned the esm usage.

Rationality: `@eslint/json` has also published cjs
![Image](https://github.com/user-attachments/assets/e829d1ac-323b-4608-9719-7f5da4d253d2)

**Comment by fasttime** [comment]
Makes sense to me :+1:

**Comment by nzakas** [comment]
This isn't possible because `mdast-util-from-markdown` is ESM-only. It doesn't bundle correctly.

**Comment by fasttime** [comment]
> This isn't possible because `mdast-util-from-markdown` is ESM-only. It doesn't bundle correctly.

That's too bad. It's another reason for supporting async parsers.

**Comment by nzakas** [comment]
Yeah, although honestly, I think we're at the point where CommonJS just isn't that important anymore. Especially with Node v22 allowing `require(esm)`, we're moving further away from it being useful to export a separate CommonJS entrypoint.

**Comment by fasttime** [comment]
> Yeah, although honestly, I think we're at the point where CommonJS just isn't that important anymore. Especially with Node v22 allowing `require(esm)`, we're moving further away from it being useful to export a separate CommonJS entrypoint.

It's true, `require("@eslint/markdown")` works in a CommonJS config file in Node.js 22.12.0:

```js
const { default: markdown } = require("@eslint/markdown");

module.exports = [
    ...markdown.configs.recommended
];
```
